### PR TITLE
Anchor-following guidance: path mode with mode-forming t-window (multimodal generation)

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Summary

Redesigns `anchor_following` guidance so a commanded prototype path can actually steer the diffusion sampler between trajectory modes, and wires it into batched RSFT generation.

- **`anchor_following` is now path following, full stop**: nearest-point Huber distance to the (extrapolated) prototype polyline, time-compensated (alpha/sigma^2) so the x0 nudge per solver step is constant, active across the mode-forming high-noise steps (inclusive of the solver's first evaluation at t=1.0). **The legacy per-timestep squared-distance ("waypoint") behavior was REMOVED** — the scale sweep below showed it has no usable operating point; configs passing `params["mode"]` fail loudly.
- **`build_path_prototypes`**: arc-length path clustering for the prototype library (speed-free shapes), replacing the speed-dominated raw-xy clustering for this use; plus an arc-aware anchor-obedience metric and the `diag_anchor_multimodality` diagnostic tool.
- **`rsft_v2_anchor` generation variant**: anchor-commanded slots in batched RSFT generation via `prototypes_path` config; `prototypes_path` is forwarded through the eval/parity tools (viz_p4_recovery, recovery_test, viz_cl_recovery, filter_avoidance_fittable, build_repaired_targets) and the lifelong runner.
- **Committed prototype libraries replaced**: both `prototypes_k16.npy` files are now (16, 40, 2) arc-length path prototypes; the raw-xy generator script and counts file are deleted; `ensure_prototypes` regenerates via `build_path_prototypes` and fails loudly on generation failure. All existing consumers (GRPO sampler anchor pool, guidance GUI, DPO annotation GUI, scene branch editor, trajectory ranker) pick the new library up through their existing defaults.
- **Rides along**: reproducer test stubs fixed to match the model turn-indicator contract (`[B, C]` logit + real rolling `turn_hist`) — those 3 tests were failing on tier4-main.

## Evidence (diag tool, zero initial noise, K=16 prototypes, 100 scenes / 8-scene sweep)

Zero noise means all output diversity comes from the guidance alone.

- **Path mode (scale 3.0)**: median endpoint spread ~14 m across commanded anchors, fans stay in the map corridor (lane/exit selection at intersections, lane/offset selection on straights); det plan unaffected.
- **Legacy waypoint mode has no usable operating point** (scale sweep 0.1→3.0): near-inert at low scale (spread ≈ noise, obedience ≈ chance); at scale >= 1.0 obedience rises to ~55% but by pasting the prototype shape over the scene — trajectories cross road borders on a straight multi-lane road. It interpolates between "no effect" and "off-road", never through scene-aware mode selection.

## Testing

- `rlvr/test_anchor_guidance.py` (new) + `rlvr/test_generation_variants.py`: 24 passed.
- Diagnostic renders/reports generated with `rlvr.autoresearch.tools.diag_anchor_multimodality` (normalizes inputs via the model's observation normalizer).

### CODE REVIEW

File: diffusion_planner/diffusion_planner/model/guidance/anchor_following.py [hunk 1]
```diff
@@ -23,39 +47,127 @@ def _load_prototypes(path: str) -> np.ndarray:
...
+        seg = anchor[1:] - anchor[:-1]
+        seg_len = seg.norm(dim=-1)
+        valid = seg_len > 0.05
+        if valid.any():
+            last = int(torch.nonzero(valid)[-1])
+            direction = seg[last] / seg_len[last]
+        else:
+            direction = torch.tensor([1.0, 0.0])
+        tail = anchor[-1] + direction * self._extend_len
+        return torch.cat([anchor, tail.unsqueeze(0)], dim=0)
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply diffusion_planner/diffusion_planner/model/guidance/anchor_following.py:134]: Fixed in 3462f561: nonzero(..., as_tuple=True)[0][-1].item() + anchor.new_tensor for the fallback direction.

File: rlvr/autoresearch/tools/build_path_prototypes.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    wheel (setup.py only packages diffusion_planner.*), so it cannot be imported
+    normally; load it from its file location next to the installed package.
+    """
+    import diffusion_planner
+
+    src = Path(diffusion_planner.__file__).parent.parent / "sampling" / "build_prototypes.py"
+    spec = importlib.util.spec_from_file_location("_dp_build_prototypes", src)
+    mod = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(mod)
+    return mod.torch_kmeans
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/build_path_prototypes.py:56]: Fixed in 3462f561: explicit FileNotFoundError/ImportError with actionable messages (source checkout / editable install).

File: docs/guidance_gui_design.md [hunk 3]
```diff
@@ -61,7 +61,14 @@ Prototypes are extracted from `ego_agent_future` (80, 3) → first 2 columns (x,
 
 ### 3.3 Prototype Generation Script
 
-File: `guidance_gui/scripts/generate_prototypes.py`
+> **Historical note:** the raw-xy generator below was REMOVED. Build prototype
+> libraries with `python -m rlvr.autoresearch.tools.build_path_prototypes`
+> (arc-length, speed-free path clustering; output `(K, M, 2)`), and see
+> `diffusion_planner/model/guidance/anchor_following.py` for the current
+> path-following energy (nearest-point Huber, time-compensated) — the
+> per-timestep squared-distance formula in this design doc no longer exists.
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply docs/guidance_gui_design.md]: Fixed in 3462f561: the note now marks the entire section (and the doc's per-timestep energy description) as historical, with the commands kept only as a design record.

File: diffusion_planner/diffusion_planner/model/guidance/anchor_following.py [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+time-compensated (multiplied by alpha_t / sigma_t^2) so the effective x0
+nudge per solver step is constant across the whole denoising schedule, and
+the guidance window covers the mode-forming high-noise steps — this is what
+makes the guidance able to actually switch trajectory modes.
+
+The legacy per-timestep squared-distance behaviour ("waypoint mode") was
+REMOVED: a scale sweep showed it has no usable operating point — near-inert
+at low scale, and at scales high enough to steer it pastes the prototype
+shape over the scene regardless of the map. Configs passing params["mode"]
+fail loudly.
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply diffusion_planner/diffusion_planner/model/guidance/anchor_following.py:18]: PR description updated: waypoint mode was removed (scale sweep showed no usable operating point); params['mode'] fails loudly. External docs (guidance design doc, rlvr README, GUI README) updated in the same PR.

File: rlvr/trajectory_ranker_gui.py [hunk 5]
```diff
@@ -252,20 +249,18 @@ def ensure_prototypes(npz_list_path: str, prototypes_path: str, force: bool = Fa
...
+                "rlvr.autoresearch.tools.build_path_prototypes",
+                "--data_list",
                 npz_list_path,
                 "--output",
                 prototypes_path,
-                "--k",
+                "--num_clusters",
                 "16",
                 "--max_samples",
                 "50000",
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/trajectory_ranker_gui.py:266]: Fixed in 3462f561: ensure_prototypes now raises on generation failure, timeout, or missing output instead of returning None into consumers.

File: rlvr/trajectory_ranker_gui.py [hunk 6]
```diff
@@ -252,20 +249,18 @@ def ensure_prototypes(npz_list_path: str, prototypes_path: str, force: bool = Fa
     if not force and Path(prototypes_path).exists():
         print(f"Using existing prototypes: {prototypes_path}")
         return prototypes_path
-    if not _GENERATE_SCRIPT.exists():
-        print(f"Warning: generate_prototypes.py not found at {_GENERATE_SCRIPT}")
-        return prototypes_path
-    print(f"Generating prototypes from {npz_list_path} -> {prototypes_path} ...")
+    print(f"Generating path prototypes from {npz_list_path} -> {prototypes_path} ...")
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/trajectory_ranker_gui.py:252]: Already fixed in 27f1037b (the internal round-2 review caught the same thing): the annotation is now `-> str`.
